lib/property_sets.rb

```diff
@@ -9,20 +9,21 @@
 
 module PropertySets
   def self.ensure_property_set_class(association, owner_class_name)
-    const_name = "#{owner_class_name}#{association.to_s.singularize.camelcase}".to_sym
+    const_name = "#{owner_class_name.demodulize}#{association.to_s.singularize.camelcase}"
+    namespace = owner_class_name.deconstantize.safe_constantize || Object
 
-    unless Object.const_defined?(const_name)
+    unless namespace.const_defined?(const_name, false)
```

Comment:
Yes - without the the `false` parameter, the `const_defined?` method will check all ancestors for whether the constant is defined, which is not the behaviour that we want.

Consider the following:
```
class Account
end

module SomeNamespace
  class Account
  end
end

module SomeOtherNamespace
end

irb(main):017:0> Object.const_defined?(:Account)
=> true
irb(main):018:0> Object.const_defined?(:Account, false)
=> true
irb(main):019:0> SomeNamespace.const_defined?(:Account)
=> true
irb(main):020:0> SomeNamespace.const_defined?(:Account, false)
=> true
irb(main):021:0> SomeOtherNamespace.const_defined?(:Account) # The constant 'Account' is not defined in SomeOtherNamespace, but because it is defined in Object this returns true.
=> true
irb(main):022:0> SomeOtherNamespace.const_defined?(:Account, false) # This correctly returns false as the constant 'Account' is not defined in 'SomeOtherNamespace'.
=> false
```

Reply:
Thanks for explaining, now it makes more sense to me. And previously, we of course didn’t care whether we looked at all ancestors since we were looking in the outermost scope anyway.